### PR TITLE
feat: added support for no-server-header and no-date-header

### PR DESCRIPTION
**Pull request by iudeen**
I have added keyword arguments to not add server header and date header. 

This fixes issue discussed in https://github.com/aaugustin/websockets/issues/1193

Please let me know if this is okay. I will add necessary test and docs.

### CODE REVIEW

File: src/websockets/server.py [hunk 1]
```diff
@@ -75,6 +75,8 @@ def __init__(
         state: State = CONNECTING,
         max_size: Optional[int] = 2**20,
         logger: Optional[LoggerLike] = None,
+        no_server_header: bool = False,
+        no_date_header: bool = False,
```
[the remark this answers is not in the document]
Comment (iudeen) [reply src/websockets/server.py:79]: okay!

File: src/websockets/server.py [hunk 2]
```diff
@@ -463,15 +469,17 @@ def reject(
...
-                ("Date", email.utils.formatdate(usegmt=True)),
+        headers_list = [
                 ("Connection", "close"),
                 ("Content-Length", str(len(body))),
                 ("Content-Type", "text/plain; charset=utf-8"),
-                ("Server", USER_AGENT),
             ]
-        )
+        if not self.no_server_header:
+            headers_list.append(("Server", USER_AGENT))
```
[the remark this answers is not in the document]
Comment (iudeen) [reply src/websockets/server.py]: Done!

**Comment by iudeen** [comment]
Added docs, tests to my best. Please let me know if there are gaps.

**Comment by aaugustin** [comment]
Thank you, I am polishing your PR in #1210.

**Comment by aaugustin** [comment]
I made minor changes, squashed, and merged in 2a07325c. Thank you for your contribution!

**Pull request closed without merging**